packages/grid/x-grid/src/tests/events.XGrid.test.tsx

```diff
@@ -303,13 +312,49 @@ describe('<XGrid /> - Events Params', () => {
 
     const { container } = render(
       <div style={{ width: 300, height: 300 }}>
-        <XGrid columns={data.columns} rows={data.rows} onRowsScrollEnd={handleOnRowsScrollEnd} />
+        <XGrid columns={data.columns} rows={data.rows} onRowsScrollEnd={handleRowsScrollEnd} />
       </div>,
     );
     const gridWindow = container.querySelector('.MuiDataGrid-window');
     // arbitrary number to make sure that the bottom of the grid window is reached.
     gridWindow.scrollTop = 12345;
     gridWindow.dispatchEvent(new Event('scroll'));
-    expect(handleOnRowsScrollEnd.callCount).to.equal(1);
+    expect(handleRowsScrollEnd.callCount).to.equal(1);
+  });
+
+  it('call onViewportRowsChange when the viewport rows change', async () => {
+    const headerHeight = 40;
+    const rowHeight = 40;
+    const pageSize = 4;
+    const handleViewportRowsChange = spy();
+    const { container } = render(
+      <div style={{ width: 300, height: headerHeight + rowHeight * pageSize }}>
+        <TestVirtualization rowHeight={rowHeight} onViewportRowsChange={handleViewportRowsChange} />
```

Comment:
Shouldn't we set the header height?

```suggestion
        <TestVirtualization headerHeight={headerHeight} rowHeight={rowHeight} onViewportRowsChange={handleViewportRowsChange} />
```